src/Processors/Formats/Impl/AvroRowOutputFormat.cpp

```diff
@@ -391,20 +391,27 @@ AvroRowOutputFormat::AvroRowOutputFormat(
 
 AvroRowOutputFormat::~AvroRowOutputFormat() = default;
 
-void AvroRowOutputFormat::writePrefix()
+void AvroRowOutputFormat::createFileWriter()
 {
-    // we have to recreate avro::DataFileWriterBase object due to its interface limitations
     file_writer_ptr = std::make_unique<avro::DataFileWriterBase>(
         std::make_unique<OutputStreamWriteBufferAdapter>(out),
         serializer.getSchema(),
         settings.avro.output_sync_interval,
         getCodec(settings.avro.output_codec));
+}
+
+void AvroRowOutputFormat::writePrefix()
+{
+    // we have to recreate avro::DataFileWriterBase object due to its interface limitations
+    createFileWriter();
 
     file_writer_ptr->syncIfNeeded();
 }
 
 void AvroRowOutputFormat::write(const Columns & columns, size_t row_num)
 {
+    if (!file_writer_ptr)
+        createFileWriter();
```

Comment:
It's ok itself, but means `writePrefix` was not called. I suppose it means we should fix call site also.
At least it was required to call `writePrefix` before, wasn't it changed?